Azure/open-service-broker-azure
# sqldb module enhancement

Due to some trouble located in https://help.github.com/articles/what-happens-to-forks-when-a-repository-is-deleted-or-changes-visibility/#changing-a-public-repository-to-a-private-repository, re-submit https://github.com/Azure/open-service-broker-azure/pull/111 from my new-forked repo.

### Changelog

* Azure SDK for GO
  * Update from v10.3.1-beta to v11.1.0-beta

* sqldb
  * Provisioning
    * Allow to enable [TDE](https://docs.microsoft.com/en-us/sql/relational-databases/security/encryption/transparent-data-encryption-azure-sql)
  * Updating
    * Allow to change service plan

### Test

#### Provisioning
Covered by lifecycle.

#### Updating
**Test updating service plan with CLI**:

1. create a sqldb instance with Basic plan

```
$ go run contrib/cmd/cli/*.go -H localhost -u username -P password provision -sid fb9bc99e-0aa9-11e6-8a8a-000d3a002ed5 -pid 3819fdfa-0aaa-11e6-86f4-000d3a002ed5 --param location=eastus --poll

Provisioning service instance 720ea58a-5dab-4b3b-b40e-2d7b3b689464

.........................

Service instance 720ea58a-5dab-4b3b-b40e-2d7b3b689464 has been successfully provisioned
```

2. update the instance with S0 plan

```
$ go run contrib/cmd/cli/*.go -H localhost -u username -P password update -sid fb9bc99e-0aa9-11e6-8a8a-000d3a002ed5 -pid 2497b7f3-341b-4ac6-82fb-d4a48c005e19 -iid 720ea58a-5dab-4b3b-b40e-2d7b3b689464 --poll

Updating service instance 720ea58a-5dab-4b3b-b40e-2d7b3b689464

............

Service instance 720ea58a-5dab-4b3b-b40e-2d7b3b689464 has been successfully updated
```

3. Check the pricing tier on Azure Portal

    The pricing tier is S0.


Ps. Is it time to add updating test to lifecycle?
